backend/accounts/backends.py

```diff
@@ -42,14 +42,21 @@ def authenticate(self, request, remote_user, shibboleth_attributes):
         )
 
         # Add initial attributes on first log in
+
         if created:
-            email = self.get_email(remote_user)
-            if email is None or email == "":
-                email = f"{shibboleth_attributes['username']}@upenn.edu"
             user.set_unusable_password()
             user.save()
             user = self.configure_user(request, user)
-            Email.objects.create(user=user, value=email, primary=True, verified=True)
+
+        # Always update email
+
+        email = self.get_email(remote_user)
+        if email is None or email == "":
```

Comment:
Would it make more sense to add a checker for if email has changed, and if it is different then update it?

Reply:
changed to reflect this